### DISCUSSION TITLE
Adding multiple cloned layers

### DISCUSSION THREAD

**Discussion by Psijic** [mapsforge/vtm#1069] [Q&A] [answered]
I didn't find in the code (not sure) if duplicated layers will not be added to the map layers. Like
```
val markerLayer = ItemizedLayer(mMap, markers, markerRendererFactory, listener)
repeat(10) {mMap.layers().add(markerLayer)}
```
adds `markerLayer` only once and there is no need to such observes. Vtm uses `Set` for it or something alike, right? (Not sure it will work this way because `ItemizedLayer` is not a `data class`).
Otherwise it should be manually restricted in Composables.

**Comment by devemux86** [comment]
You can check the various samples.
They have examples where many layers are created in a loop and added on the map.

**Reply by Psijic** [reply]
Suppose, their layers were different. I could clone and repeat 1000 layers for performance test but not sure it solve the problem of duplicating even if performance is ok.

**Reply by devemux86** [reply, marked as answer]
The procedure is to create and add many layers (e.g. PathOverlayActivity).
For layers that host multiple items, create one or many layers and fill them with items (e.g. MarkerOverlayActivity, VectorLayerActivity).

**Comment by Psijic** [comment]
The case is next: geohash is used to show only current map position POI markers. If I remove and add a new layer - I see blinks. If I don't remove the layer - markers duplicates or their count grows as much to hit the performance.
